gammapy/background/tests/test_models.py

```diff
@@ -43,3 +47,85 @@ def test_y_model(self):
         model = self.model.y_model(-30)
         assert isinstance(model, Gaussian1D)
         assert_allclose(model.parameters, [0, -1, 0.4])
+
+class TestCubeBackgroundModel():
+
+##    def test_read(self):
+##        # test shape of bg cube when reading a file
+##        filename = 'data/bg_test.fits'
+##        #DIR = '/home/mapaz/astropy/development_code/gammapy/gammapy/background/tests/'
+##        #filename = DIR + filename
+##        filename = get_pkg_data_filename(filename)
+##        # TODO: this is failing!!!
+##        # maybe because the class is not part of the module?!!!
+##        # Here it works:
+##        # gammapy/irf/tests/test_effective_area.py test_EffectiveAreaTable
+##        # TODO: the test file doesn't have units for the det x,y axes!!!
+##        #        produce correct file (test_write), then come back here!!!
+##        bg_cube_file = CubeBackgroundModel.read(filename)
+##        assert len(bg_cube_file.background.shape) == 3
+
+
+    def test_image_plot(self):
+
+        DIR = '/home/mapaz/astropy/testing_cube_bg_michael_mayer/background/'
+        filename = 'hist_alt3_az0.fits.gz'
+        filename = DIR + filename
+        #TODO: change this, when test_read is fixed!!!
+        #      - use data/bg_test.fits
+        #      - use get_pkg_data_filename
+        bg_cube_model = CubeBackgroundModel.read(filename)
+
+        # test bg rate values plotted for image plot of energy bin conaining E = 2 TeV
+        energy = Quantity(2., 'TeV')
+        fig_image, ax_im, image_im = bg_cube_model.plot_images(energy)
+        plot_data = image_im.get_array()
+
+        # get data from bg model object to compare
+        energy_bin, energy_bin_edges = bg_cube_model.find_energy_bin(energy)
+        model_data = bg_cube_model.background[energy_bin]
+
+        # test if both arrays are equal
+        decimal = 4
+        np.testing.assert_almost_equal(plot_data, model_data.value, decimal)
+
+
+    def test_spectrum_plot(self):
+
+        DIR = '/home/mapaz/astropy/testing_cube_bg_michael_mayer/background/'
+        filename = 'hist_alt3_az0.fits.gz'
+        filename = DIR + filename
+        #TODO: change this, when test_read is fixed!!!
+        #      - use data/bg_test.fits
+        #      - use get_pkg_data_filename
+        bg_cube_model = CubeBackgroundModel.read(filename)
+
+        # test bg rate values plotted for spectrum plot of detector bin conaining det (0, 0) deg (center)
+        det = Angle([0., 0.], 'degree')
+        fig_spec, ax_spec, image_spec = bg_cube_model.plot_spectra(det)
+        plot_data = ax_spec.get_lines()[0].get_xydata()
+
+        # get data from bg model object to compare
+        det_bin, det_bin_edges = bg_cube_model.find_det_bin(det)
+        model_data = bg_cube_model.background[:, det_bin[0], det_bin[1]]
+
+        # test if both arrays are equal
+        decimal = 4
+        np.testing.assert_almost_equal(plot_data[:,1], model_data.value, decimal)
+
+
+    def test_write(self):
```

Comment:
Here's examples how to generate a temp file name from Gammapy tests to use:
https://github.com/gammapy/gammapy/search?utf8=%E2%9C%93&q=tempfile